scripts/Stretch.py

```diff
@@ -0,0 +1,419 @@
+        """
+        dmin_tri = self.line_width / 2.0
+        iextra = np.floor_divide(len(orig_seq), 3) # Nb of extra points
+        ibeg = 0 # Index of first point of the triangle
+        iend = 0 # Index of the third point of the triangle
+        for i, step in enumerate(orig_seq):
+            # i is the index of the second point of the triangle
+            # pos_after is the array of positions of the original sequence
+            # after the current point
+            pos_after = np.resize(np.roll(orig_seq, -i-1, 0), (iextra, 2))
+            # Vector of distances between the current point and each following point
+            dist_from_point = ((step - pos_after) ** 2).sum(1)
+            if np.amax(dist_from_point) < dmin_tri * dmin_tri:
+                continue
+            iend = np.argmax(dist_from_point >= dmin_tri * dmin_tri)
+            # pos_before is the array of positions of the original sequence
+            # before the current point
+            pos_before = np.resize(np.roll(orig_seq, -i, 0)[::-1], (iextra, 2))
+            # This time, vector of distances between the current point and each preceding point
+            dist_from_point = ((step - pos_before) ** 2).sum(1)
+            if np.amax(dist_from_point) < dmin_tri * dmin_tri:
+                continue
+            ibeg = np.argmax(dist_from_point >= dmin_tri * dmin_tri)
+            # See https://github.com/electrocbd/post_stretch for explanations
+            # relpos is the relative position of the projection of the second point
+            # of the triangle on the segment from the first to the third point
+            # 0 means the position of the first point, 1 means the position of the third,
+            # intermediate values are positions between
+            length_base = ((pos_after[iend] - pos_before[ibeg]) ** 2).sum(0)
+            relpos = ((step - pos_before[ibeg])
+                      * (pos_after[iend] - pos_before[ibeg])).sum(0)
+            if np.fabs(relpos) < 1000.0 * np.fabs(length_base):
+                relpos /= length_base
+            else:
+                relpos = 0.5 # To avoid division by zero or precision loss
+            projection = (pos_before[ibeg] + relpos * (pos_after[iend] - pos_before[ibeg]))
+            dist_from_proj = np.sqrt(((projection - step) ** 2).sum(0))
+            if dist_from_proj > 0.001: # Move central point only if points are not aligned
+                modif_seq[i] = (step - (self.stretch / dist_from_proj)
+                                * (projection - step))
+        return
+
+    def wideTurn(self, orig_seq, modif_seq):
+        '''
+        We have to select three points in order to form a triangle
+        These three points should be far enough from each other to have
+        a reliable estimation of the orientation of the current turn
+        '''
+        dmin_tri = self.line_width / 2.0
+        ibeg = 0
+        i = 1
+        iend = 2
+        while i+1 < len(orig_seq):
```

Comment:
Again, it should be possible to use `for i in range(1, len(orig_seq) - 1):`

Then you wouldn't need to track `good_triangle` any more either, since you can just use `continue` instead of `good_triangle = False` if there's no increment of `i` at the end of the loop.